src/azul/service/responseobjects/hca_response_v5.py

```diff
@@ -253,12 +255,22 @@ def _push_content(self) -> str:
                 if self.format == 'tsv':
                     self._write_tsv(text_buffer)
                 elif self.format == 'full':
-                    self._write_full(text_buffer)
+                    project_short_name = self._write_full(text_buffer)
+                    self._push_name_file(object_key, project_short_name)
                 else:
                     raise NotImplementedError(f'Multipart upload not implemented for `{self.format}`')
 
         return object_key
 
+    def _push_name_file(self, object_key, project_short_name):
+        project_short_name = project_short_name or ''
+        file_name_prefix = unicodedata.normalize('NFKD', project_short_name)
+        file_name_prefix = re.sub(r'[^\w ,.@%&\-_()\\[\]/{}]', '_', file_name_prefix).strip()
+        timestamp = datetime.now().strftime("%Y-%m-%d %H.%M")
+        filename = f'{file_name_prefix} {timestamp}.tsv'.strip()
```

Comment:
It looks like if `project_short_name` is None or False or empty etc, that we'd end up with a file name that starts with a space. I think that we only want to create a `.name` object if we have a project name.

Reply:
So @hannes-ucsc if a project_short_name is None or False or empty etc, does the tsv filename default to `hca-manifest-5d67606f-472a-4bb9-9dac-5135ef5c164c.tsv` (if multiple part manifest are enabled)?

Reply:
The manifest object's key should be the same as it is now, prior to this change. Only if all rows in the TSV refer to exactly one `project_short_name` AND if that `project_short_name` is not empty, will the `.name` object be created. If a `.name` object exists when the signed URL is created, the CD header's fileName element will be set to the contents of the `.name` object. This will happen independently of whether single or multipart upload was enabled.

Reply:
Rather than silently stripping the leading space that would result from an empty `project_short_name`, assert that it is not empty.

Geryl, I beg you, please break this habit of writing lenient code that masks errors. Be strict.